src/runner.ts

```diff
@@ -17,36 +17,29 @@ const isStdInSource = (inventory: DocumentInventory): boolean => {
   return inventory.document.source === STDIN;
 };
 
-const generateDefinedExceptionsButStdIn = (documentInventory: DocumentInventory): IRuleResult => {
-  return generateDocumentWideResult(
-    documentInventory.document,
-    'The ruleset contains `except` entries. However, they cannot be enforced when the input is passed through stdin.',
-    DiagnosticSeverity.Warning,
-    'except-but-stdin',
-  );
-};
+export interface IRunningContext {
+  documentInventory: DocumentInventory;
+  rules: RunRuleCollection;
+  functions: FunctionCollection;
+  exceptions: RulesetExceptionCollection;
+}
 
-export const runRules = (
-  documentInventory: DocumentInventory,
-  rules: RunRuleCollection,
-  functions: FunctionCollection,
-  exceptions: RulesetExceptionCollection,
-): IRuleResult[] => {
-  const results: IRuleResult[] = [];
+export const runRules = async (context: IRunningContext): Promise<IRuleResult[]> => {
+  const { documentInventory, rules, exceptions } = context;
 
+  const results: Array<Promise<IRuleResult[]>> = [];
   const isStdIn = isStdInSource(documentInventory);
+  const exceptRuleByLocations = isStdIn ? {} : pivotExceptions(exceptions, rules);
 
   if (isStdIn && Object.keys(exceptions).length > 0) {
-    results.push(generateDefinedExceptionsButStdIn(documentInventory));
+    // todo: use @stoplight/reporter
+    console.warn(
```

Comment:
@P0lip Why change this? Having those "you're using spectral wrong" messages as results is pretty useful when you don't use the cli as is (or just leveraging a reporter).

Reply:
We would still print out to the terminal :slightly_frowning_face: 
What we would gain is the ability to control the output more granularly, i.e. based on the value of `verbose` flag.

Reply:
>We would still print out to the terminal 🙁

Indeed, but that doesn't provide the user with a knob to pontially choose to fail a CI build when Spectral is used wrong.

Reply:
Ahhhh. That's what you mean. Sorry, misunderstood it initially. Yes, you are right. I'll revert it